### PR TITLE
Add chatops generator

### PR DESCRIPTION
**Pull request by joshuatobin**
Just a simple generator stub for creating chatops commands.

CC: @lexelby @reidmix

### CODE REVIEW

**Comment by lexelby** [comment]
That's so cool!! Can we have it generate a spec too?

**Comment by joshuatobin** [comment]
@lexelby ha yeah. I have that just forgot to push it up.

File: lib/generators/chatops/chatops_generator.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  end
+
+  def copy_spec
+    create_file "spec/lib/chat_ops/commands/#{file_name}_command_spec.rb", <<-FILE
+RSpec.describe ChatOps::Commands::AddResponderCommand do
+  include ChatOpsCommandHelper
+
+  describe 'regex' do
+    test_regex_against_commands <<-EOL
+      # some command
```
Comment (lexelby) [inline lib/generators/chatops/chatops_generator.rb]: Perhaps add a second line, `# some other command` to indicate that commands can be separated by newlines?

File: lib/generators/chatops/chatops_generator.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    EOL
+  end
+
+  describe '.run' do
+    it "tests some thing" do
```
Comment (lexelby) [inline lib/generators/chatops/chatops_generator.rb]: perhaps add a line showing how `process` is used?

**Comment by joshuatobin** [comment]
@lexelby added suggestions

**Comment by lexelby** [comment]
Awesome.  Shipit :+1:

File: lib/generators/chatops/USAGE [hunk 3]
```diff
@@ -0,0 +1,7 @@
+Description:
+    Generator for creating new chatops commands
+
+Example:
+    rails generate chatops COMMAND
+
+    This will stub out a command file in lib/chatops/commands
```
Comment (reidmix) [inline lib/generators/chatops/USAGE]: Cool, I didn't know about `USAGE`

File: lib/generators/chatops/chatops_generator.rb [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  class #{class_name}Command < ChatOps::ChatOpsCommand
+    match //
+    help_message ""
+
+    def run(user, match)
+      # ChatOps.message()
+    end
+  end
+end
+    FILE
```
Comment (reidmix) [inline lib/generators/chatops/chatops_generator.rb]: You can also put this in a template in the directory found on line 2.  

Then use the template command like here: https://github.com/heroku/loggerator/blob/master/lib/generators/loggerator/log_generator.rb#L9

You'd need to convert the template to `erb`

Anyway, that'd help separate your files from your code which cleans things up nicely.

File: lib/generators/chatops/chatops_generator.rb [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    EOL
+  end
+
+  describe '.run' do
+    it "tests some thing" do
+      # expect(process("add @#{user1.handle} to incident")).to react_with(':checkmark:').and not_have_message
+    end
+  end
+end
+    FILE
```
Comment (reidmix) [inline lib/generators/chatops/chatops_generator.rb]: Same suggestion using templates.

File: lib/generators/chatops/chatops_generator.rb [hunk 6]
```diff
@@ -0,0 +1,39 @@
+class ChatopsGenerator < Rails::Generators::NamedBase
+  source_root File.expand_path('../templates', __FILE__)
+
+  def copy_file
+    create_file "lib/chat_ops/commands/#{file_name}_command.rb", <<-FILE
+module ChatOps::Commands
+  class #{class_name}Command < ChatOps::ChatOpsCommand
+    match //
```
Comment (reidmix) [inline lib/generators/chatops/chatops_generator.rb]: maybe put a little comment to show this is a regex
`match // # add your regex to match here` kind of thing

File: lib/generators/chatops/chatops_generator.rb [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    help_message ""
```
Comment (reidmix) [inline lib/generators/chatops/chatops_generator.rb]: same thing, add a comment to indicate what the string should be
`help_message "" # add your usage here` kind of thing

**Pull request merged**